client/replays/replay-drop.tsx

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useEffect } from 'react'
+import { useAppDispatch } from '../redux-hooks'
+import { openReplay } from './action-creators'
+
+export default function ReplayDrop() {
+  const dispatch = useAppDispatch()
+
+  const onDrop = useCallback(
+    (e: DragEvent) => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      const [file] = e.dataTransfer?.files || []
+      if (file?.path.endsWith('.rep')) {
+        dispatch(openReplay(file.path))
+      }
+    },
+    [dispatch],
+  )
+
+  const onDragOver = useCallback((e: DragEvent) => {
+    e.preventDefault()
+    e.stopPropagation()
+
+    // NOTE: it is not possible to access files during `dragover` now,
+    // see this electron issue:
+    // https://github.com/electron/electron/issues/9840
+    // Thus, just allow any files here and deal with the garbage
+    // in `drop` handler
+    if (e.dataTransfer) e.dataTransfer.dropEffect = 'link'
+  }, [])
+
+  useEffect(() => {
+    document.addEventListener('drop', onDrop)
```

Comment:
I don't really like that these events are being attached to the document, as I don't think context-less handling for replays is very likely to last that long. We definitely want to be able to e.g. drag map files in to upload and play on, probably drag replay files into the party interface to watch replays with others, etc. To do all that, we're much better served by contextual event handling, rather than one handler at the root.

A better approach that will serve us long-term here would be to make a component that fills its container completely, and attach the drag handlers to that. This would also give us a place to show a UI as you're dragging, which will actually make this discoverable.